theano/gpuarray/dnn.py

```diff
@@ -3130,6 +3132,66 @@ def local_dnn_reduction(node):
                             False)(node.inputs[0]),)
 
 
+@register_opt('cudnn')
+@local_optimizer([GpuMaxAndArgmax])
+def local_cudnn_maxandargmax(node):
+    if not isinstance(node.op, GpuMaxAndArgmax):
+        return
+
+    if not dnn_available(node.inputs[0].type.context_name):
+        return
+
+    if version(raises=False) < 6000:
+        return
+
+    if node.inputs[0].ndim > 8:
+        return
+
+    if node.inputs[0].dtype != node.outputs[0].dtype:
+        return
+
+    if node.inputs[0].dtype not in ['float16', 'float32', 'float64']:
+        return
+
+    # order of the axes influences the output indices
```

Comment:
I am pretty sure that axis is sorted on the CPU so this check should always pass.  But I would like to keep this to make sure we don't return bad results.